### PR TITLE
Support additional LDAP Organization fields

### PR DESCRIPTION
**Pull request by cmangeat**
Proposal for #2146

### CODE REVIEW

**Comment by fvanderbiest** [comment]
There are conflicts, can you update the PR please ?

**Comment by jeanpommier** [comment]
You're keeping the competency area in the groupOfMembers' description field ? Woudn't it be more explicit to use a dedicated field in the georchestraOrg object ?

**Comment by fvanderbiest** [comment]
> You're keeping the competency area (list of INSEE IDs) in the groupOfMembers' description field ? Woudn't it be more explicit to use a dedicated field in the georchestraOrg object ?

Very good point Jean.
We should indeed take advantage of this PR to clean the mess, eventually creating a new schema field.

**Comment by landryb** [comment]
Yes, but once and for all... think of the ones who built code that uses the existing DIT schema :)

**Comment by fvanderbiest** [comment]
> Yes, but once and for all... think of the ones who built code that uses the existing DIT schema :)

You're a bit late to the game... 
I'm not sure I get your point. Would you like us to not use custom fields ?

**Comment by jeanpommier** [comment]
No, I think Landry says not to move this extent information elsewhere; We actually can keep it on the description field. It is just not very obvious which is which, since we'll have a description field on both objects defining the organization.
Anyway, Id' tend to say that if we move it in a dedicated field in the georchestraOrg object, it should not be so difficult to adjust existing apps to use it, right ?
And yet, will your applications support the pending evolution in the LDAP schema ? If not, it will indeed be a good opportunity to clean up a little bit

**Comment by landryb** [comment]
Its fine to use a custom schema instead of abusing existing object fields, and i dont mind if description content is moved to another attr, as long as it's changing at every release.

adding attributes is fine (like for image/url/etc), its renaming/repurposing attributes that can get hairy.

**Comment by cmangeat** [comment]
@groldan here...

**Comment by cmangeat** [comment]
https://github.com/georchestra/georchestra/commit/816c170bc9e01dd21003fbdaeccd34a6b35c86ba need the "right" latest docker image to be available for tests.

**Comment by groldan** [comment]
@cmangeat please rebase on top of master, the branch 60 commits behind master.
It rebases fine, but when running `mvn verify -Pit` to run integration tests, get a bunch of failures like this one: 
```
[ERROR] Failures: 
[ERROR]   OrgsIT.createAndGet:62 Status expected:<200> but was:<500>
[ERROR]   OrgsIT.createAndGetWithDescription:95 No value at JSON path "$.description": com.jayway.jsonpath.PathNotFoundException: Expected to find an object with property ['description'] in path $ but found 'java.lang.String'. This is not a json object according to the JsonProvider: 'com.jayway.jsonpath.spi.json.JsonSmartJsonProvider'.
```

File: console/src/it/java/org/georchestra/console/integration/IntegrationTestSupport.java [hunk 1]
```diff
@@ -119,4 +122,10 @@ public ResultActions perform(RequestBuilder requestBuilder) throws Exception {
     public String testName() {
         return testName.getMethodName();
     }
+
+    public String readRessourceToString(String name) throws URISyntaxException, IOException {
```
[the remark this answers is not in the document]
Comment (cmangeat) [reply console/src/it/java/org/georchestra/console/integration/IntegrationTestSupport.java]: 816c170 need the "right" latest docker image to be available for tests.

[the remark this answers is not in the document]
Comment (cmangeat) [reply console/src/it/java/org/georchestra/console/integration/IntegrationTestSupport.java]: fixed... thanks.

**Comment by fvanderbiest** [comment]
Tested, looks good !

Capture du 2019-07-23 10-46-26

Capture du 2019-07-23 10-46-50

**Pull request merged**